backend/core/scripts/import-listings-from-csv.ts

```diff
@@ -0,0 +1,177 @@
+import csv from "csv-parser"
+import fs from "fs"
+import { importListing, createUnitsArray } from "./listings-importer"
+import { Listing } from "../src/listings/entities/listing.entity"
+import { Property } from "../src/property/entities/property.entity"
+import { Address } from "../src/shared/entities/address.entity"
+import { CountyCode } from "../src/shared/types/county-code"
+import { CSVFormattingType } from "../src/csv/types/csv-formatting-type-enum"
+
+// This script reads in listing data from a CSV file and sends requests to the backend to create
+// the corresponding Listings. A few notes:
+// - This script does not delete or modify any existing listings.
+// - If one listing fails to be uploaded, the script will still attempt all the rest. At the end,
+//   it will report how many failed (with error messages) and how many succeeded.
+// - Each line in the CSV file is assumed to correspond to a distinct listing.
+// - This script assumes particular heading names in the input CSV file (see listingFields["..."]
+//   below).
+
+// Sample usage:
+// $ yarn ts-node scripts/import-listings-from-csv.ts http://localhost:3100 admin@example.com:abcdef path/to/file.csv
+
+async function main() {
+  if (process.argv.length < 5) {
+    console.log(
+      "usage: yarn ts-node scripts/import-listings-from-csv.ts import_api_url email:password csv_file_path"
+    )
+    process.exit(1)
+  }
+
+  const [importApiUrl, userAndPassword, csvFilePath] = process.argv.slice(2)
+  const [email, password] = userAndPassword.split(":")
+
+  // Regexes used to parse the "Affordability Mix" field
+  const amiRangeRegex = /(\d+)-(\d+)% AMI/ // e.g. 30-60% AMI
+  const amiValueRegex = /^(\d+)% AMI/ // e.g. 40% AMI
+  const amiUpperLimitRegex = /^Up to (\d+)% AMI/ // e.g. Up to 80% AMI
+
+  // Read raw CSV data into memory.
+  // Note: createReadStream creates ReadStream's whose on("data", ...) methods are called
+  // asynchronously. To ensure that all CSV lines are read in before we start trying to upload
+  // listings from it, we wrap this step in a Promise.
+  const rawListingFields = []
+  const promise = new Promise<void>((resolve, reject) => {
+    fs.createReadStream(csvFilePath)
+      .pipe(csv())
+      .on("data", (listingFields) => {
+        // Include only listings that are "regulated" affordable housing
+        const affordabilityStatus: string = listingFields["Affordability status [Regulated Only]"]
+        if (affordabilityStatus.toLowerCase() === "regulated") {
+          rawListingFields.push(listingFields)
+        }
+      })
+      .on("end", resolve)
+      .on("error", reject)
+  })
+  await promise
+
+  console.log(`CSV file successfully read in; ${rawListingFields.length} listings to upload`)
+
+  const uploadFailureMessages = []
+  let numListingsSuccessfullyUploaded = 0
+  for (const listingFields of rawListingFields) {
+    const listing = new Listing()
+    const property = new Property()
+    const address = new Address()
```

Comment:
Could you also populate a UnitsSummary here and attach it to the property?

UnitsSummary should really be a property of a `Listing`. I'm changing that [here](https://github.com/bloom-housing/bloom/pull/1670), but that PR won't be pulled into our repo until at least next week and I don't think it should block this.

Reply:
I started working on this, then realized this is blocked by #406, since we're not able to get the correct property ID for the listing we upload :(

I created https://github.com/CityOfDetroit/bloom/issues/410 to track making this change, though. I'll submit this and will work on #406 to hopefully unblock #410.